allen-cell-animated/timelapse-colorizer
# feature: Internal build mode

Problem
=======
Closes #476, adding an internal build mode.

*Estimated review size: small, 5 minutes*

Solution
========
- Adds a second internal build config for `vite` which can be accessed by running `npm run start-internal` or `npm run build-internal`.
- Adds a second PR preview that builds the internal version. (These can be accessed by replacing the `pr-preview` folder in the URL with `pr-preview-internal`.)
- Adds a global constant that reads whether the app is in internal mode.
- Added indicators for internal mode to the header, version info, and console log.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open internal preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-477/
2. Go to the help menu and open "Version Info". You should see text stating that this is an internal build.
3. Repeat the same steps with the regular preview link. The message should not appear. https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-477/

Screenshots (optional):
-----------------------

![image](https://github.com/user-attachments/assets/c5ee1f0e-05c5-4955-a5c0-2f88c48a0d6e)

![image](https://github.com/user-attachments/assets/1184d322-c333-4738-92a9-35949bf7e165)


Keyfiles (delete if not relevant):
-----------------------
1. main file/entry point
4. other important file

Thanks for contributing!
